crates/apollo-mcp-server/src/operations.rs

```diff
@@ -1019,6 +1019,21 @@ impl graphql::Executable for Operation {
             }
         }
     }
+
+    fn operation_name(&self) -> Option<String> {
+        match Parser::new().parse_ast(&self.inner.source_text, "operation") {
+            Ok(document) => {
+                // Find the first operation definition
+                for definition in document.definitions {
+                    if let Definition::OperationDefinition(operation) = definition {
+                        return operation.name.as_ref().map(|name| name.to_string());
+                    }
+                }
+                None
+            }
+            Err(_) => None,
+        }
+    }
```

Comment:
Operation::from_document already parsed out the name so we can add `operation_name` to the `Operation` struct and return that
```suggestion
       Some(self.operation_name)
```